lib/install/inflate-shrinkwrap.js

```diff
@@ -29,7 +29,7 @@ var inflateShrinkwrap = module.exports = function (tree, swdeps, finishInflating
     if (child && (child.fromShrinkwrap ||
                   (sw.resolved && child.package._resolved === sw.resolved) ||
                   (sw.from && url.parse(sw.from).protocol && child.package._from === sw.from) ||
-                  child.package.version === sw.version)) {
+                  (!child.package.gitHead && child.package.version === sw.version))) {
```

Comment:
I'd like a more general correction here. This code is saying that a module on disk matches the shrinkwrap if ANY of these are true, and I'm pretty sure that's not actually what we want to do.

I don't know how I got my head screwed on wrong when I wrote that, but I'm ​_pretty sure_​ I intended those to be exclusive, eg, if there's a _resolved then rely on that for identity, else if there's a from with a protocol, use that for identity, and if nothing else is available use version for identity.

So I _think_ this should actually read:
```js
    if (child) {
      var useExisting
      if (child.fromShrinkwrap) {
        useExisting = true
      } else if (sw.resolved) {
        useExisting = child.package._resolved === sw.resolved
      } else if (sw.from && url.parse(sw.from).protocol) {
        useExisting = child.package._from === sw.from
      } else if (child.package.version === sw.version) {
        useExisting = true
      }
      if (useExisting) {
        if (!child.fromShrinkwrap) child.fromShrinkwrap = spec
        tree.children.push(child)
        return inflateShrinkwrap(child, sw.dependencies || {}, next)
      }
    }
```